### PR TITLE
Convert reference documentation code snippets to self contained examples/yea tests

### PR DESCRIPTION
**Pull request by dannygoldstein**
Description
-----------

Hackweek project. 

Improves documentation by converting docstring examples into yea tests 

Testing
-------

PR makes docstring examples into tests 

Release Notes
-------------

Below, please enter user-facing release notes as one or more bullet points.
If your change is not user-visible, write `NO RELEASE NOTES` instead, with no bullet points.

------------- BEGIN RELEASE NOTES ------------------
NO RELEASE NOTES
------------- END RELEASE NOTES --------------------

### CODE REVIEW

File: wandb/wandb_agent.py [hunk 1]
```diff
@@ -540,16 +540,30 @@ def agent(sweep_id, function=None, entity=None, project=None, count=None):
...
-        def train():
-            with wandb.init() as run:
-                print("config:", dict(run.config))
-                for epoch in range(35):
-                    print("running", epoch)
-                    wandb.log({"metric": run.config.param1, "epoch": epoch})
-                    time.sleep(1)
-
-        wandb.agent(sweep_id, function=train)
+       ```python
```
Comment (kptkin) [inline wandb/wandb_agent.py:543]: are you missing: `<!--yeadoc-test:...-->`?

Comment (charlesfrye) [reply wandb/wandb_agent.py]: Yep! This same test appears in two places, so this code as it stands is actually being tested by `one-parameter-sweep.yea`.

But moving forward, these two code snippets might diverge, so I've added `one-parameter-sweep-agent.yea` to track this one.

Comment (kptkin) [reply wandb/wandb_agent.py]: got ya! thanks 👍

**Review by charlesfrye — APPROVED** [review]
LGTM 🤡

**Pull request merged**